src/program/snabbvmx/lwaftr/lwaftr.lua

```diff
@@ -0,0 +1,169 @@
+      end
+   end
+   function handlers.i(arg)
+      id = arg
+      if not arg then
+         fatal("Argument '--id' was not set")
+      end
+   end
+   function handlers.p(arg)
+      pci = arg
+      if not arg then
+         fatal("Argument '--pci' was not set")
+      end
+   end
+   function handlers.m(arg)
+      mac = arg
+      if not arg then
+         fatal("Argument '--mac' was not set")
+      end
+   end
+   function handlers.s(arg)
+      sock_path = arg
+      if not arg then
+         fatal("Argument '--sock' was not set")
+      end
+   end
+   function handlers.mirror (arg)
+      mirror_id = arg
+   end
+   function handlers.h() show_usage(0) end
+   lib.dogetopt(args, handlers, "c:s:i:p:m:vD:h", {
+      ["conf"] = "c", ["sock"] = "s", ["id"] = "i", ["pci"] = "p", ["mac"] = "m",
+      ["mirror"] = 1, verbose = "v", duration = "D", help = "h" })
+   return opts, conf_file, id, pci, mac, sock_path, mirror_id
+end
+
+function run(args)
+   local opts, conf_file, id, pci, mac, sock_path, mirror_id = parse_args(args)
+
+   local conf = {}
+   local lwconf = {}
+   local ring_buffer_size = 2048
+   local discard_threshold = 100000
+   local discard_check_timer = 1
+   local discard_wait = 20
+
+   if file_exists(conf_file) then
+      conf = lib.load_conf(conf_file)
+      if not file_exists(conf.lwaftr) then
+         fatal(("lwAFTR conf file '%s' not found"):format(conf.lwaftr))
+      end
+      lwconf = require('apps.lwaftr.conf').load_lwaftr_config(conf.lwaftr)
+      lwconf.ipv6_mtu = lwconf.ipv6_mtu or 1500
+      lwconf.ipv4_mtu = lwconf.ipv4_mtu or 1460
+   else
+      print(("Interface '%s' set to passthru mode"):format(id))
+      ring_buffer_size = 1024
+      conf.settings = {}
+   end
+
+   if conf.settings then
+      if conf.settings.discard_threshold then
+         discard_threshold = conf.settings.discard_threshold
+      end
+      if conf.settings.discard_check_timer then
+         discard_check_timer = conf.settings.discard_check_timer
+      end
+      if conf.settings.discard_wait then
+         discard_wait = conf.settings.discard_wait
+      end
+      if conf.settings.ring_buffer_size then
+         ring_buffer_size = tonumber(conf.settings.ring_buffer_size)
+         if not ring_buffer_size then
+            fatal("Bad ring size: " .. conf.settings.ring_buffer_size)
+         end
+         if ring_buffer_size > 32*1024 then
+            fatal("Ring size too large for hardware: " .. ring_buffer_size)
+         end
+         if math.log(ring_buffer_size)/math.log(2) % 1 ~= 0 then
+            fatal("Ring size is not a power of two: " .. ring_buffer_size)
+         end
+      end
+   end
+
+   set_ring_buffer_size(ring_buffer_size)
+
+   if id then
+      local lwaftr_id = shm.create("nic/id", lwtypes.lwaftr_id_type)
+      lwaftr_id.value = id
+   end
+
+   local vlan = conf.settings.vlan
+
+   conf.interface = { 
+      mac_address = mac,
+      pci = pci, 
+      id = id, 
+      mtu = 9500,
```

Comment:
I'm ok with hardwiring this value, as suggested by Marcel, but please put it as a constant at the top of the file, with a meaningful name, so that it's more apparent.